WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```diff
@@ -205,6 +264,83 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
             )
         }
     }
+    
+    public func fetchConversation(
+        with id: UUID,
+        domain: String?
+    ) async -> ZMConversation? {
+        await context.perform { [context] in
+            ZMConversation.fetch(
+                with: id,
+                domain: domain,
+                in: context
+            )
+        }
+    }
+    
+    public func addSystemMessage(
```

Comment:
What do you think about creating a MessageRepository and move that method there ? It may seem a bit overkill for now but looking at Android they do have a MessageRepository that has many methods.

Reply:
Yes, it would likely be one method for each message type. We can do it in a separate PR.

Reply:
sounds good, I created a Jira [ticket](https://wearezeta.atlassian.net/browse/WPB-11839)